lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -3481,6 +3481,34 @@ bool NonLocalAnchor::updateModule(StringAttr oldMod, StringAttr newMod) {
   return updateMade;
 }
 
+bool NonLocalAnchor::updateModuleAndInnerRef(
+    StringAttr oldMod, StringAttr newMod,
+    const llvm::DenseMap<StringAttr, StringAttr> &renameMap) {
+  auto fromRef = FlatSymbolRefAttr::get(oldMod);
+  SmallVector<Attribute> namepathNew;
+  bool updateMade = false;
+  for (auto element : namepath()) {
+    if (auto innerRef = element.dyn_cast<hw::InnerRefAttr>()) {
+      if (innerRef.getModule() == oldMod) {
+        // If the leaf module updated, then the symbol should also be renamed.
+        auto to = renameMap.find(innerRef.getName());
+        assert(to != renameMap.end() && "should have been renamed");
```

Comment:
Shouldn't it not just exist be be equal to newMod?
Is there a reason this isn't just a loop testing if each module is in the rename map?  Why pass in a old and new and also a rename map?

Reply:
I think the confusion here is that, the rename map consists of inner symbols and not the module names.